assets/src/main/kotlin/ktx/assets/disposables.kt

```diff
@@ -98,3 +100,85 @@ inline fun <Asset : Disposable> Array<Asset>?.dispose(onError: (Exception) -> Un
 @Suppress("unused", "NOTHING_TO_INLINE")
 inline fun Throwable?.ignore() {
 }
+
+/**
+ * Interface describing a container for [Disposable]s that provides functions for disposing all
+ * its registered items. An implementing class's Disposable declarations can be tagged with
+ * [.alsoRegister()][alsoRegister] to conveniently register them as they are instantiated and assigned.
+ *
+ * Calling [dispose] on the registry will call [dispose] on all its registered members.
+ *
+ * The existing implementation [DisposableContainer] can be attached to a class as a delegate to
+ * attach these convenience functions to any class.
+ */
+interface DisposableRegistry : Disposable {
+
+  /**
+   * A copy of the of the registered Disposables.
+   */
+  val registeredDisposables: Set<Disposable>
+
+  /**
+   * Registers [disposable] with this registry.
+   * @return true if the item was successfully registered or false if it was already registered.
+   */
+  fun register(disposable: Disposable): Boolean
+
+  /**
+   * Removes [disposable] from this registry.
+   * @return true if the item was successfully removed or false if it was not in the registry.
+   */
+  fun deregister(disposable: Disposable): Boolean
+
+  /**
+   * Removes all disposables from the registry without disposing them.
+   * @return true if any items were in the registry.
+   */
+  fun deregisterAll(): Boolean
+
+  /**
+   * Calls [dispose][Disposable.dispose] on each registered Disposable.
+   */
+  override fun dispose()
+
+  /**
+   * Register this [Disposable] with the [DisposableRegistry].
+   * @return this same object
+   */
+  fun <T : Disposable> T.alsoRegister(): T {
+    register(this)
+    return this
+  }
+
+  /**
+   * Remove this [Disposable] from the [DisposableRegistry] if it is already registered.
+   * @return this same object
+   */
+  fun <T : Disposable> T.alsoDeregister(): T {
+    deregister(this)
+    return this
+  }
+}
+
+/**
+ * An implementation of [DisposableRegistry] that can be subclassed or attached as a delegate to
+ * extend DisposableRegistry's convenience functions to [Disposable] declarations within the class.
+ */
+open class DisposableContainer : DisposableRegistry {
+
+  private val registry: MutableSet<Disposable> = Collections.newSetFromMap(IdentityHashMap())
+
+  override val registeredDisposables: Set<Disposable> get() = registry.toSet()
```

Comment:
Note that this will not create an identity set; it will return a `LinkedHashSet` (afaik) instead. You might need to change this to `.toCollection(Collections.newSetFromMap(IdentityHashMap()))`.

Reply:
I researched this some more, and the way I did it should be fine. I wrote a test for it, and it passes that part, but `registeredDisposables` fails because `toSet()` doesn't give you an identity `Set`. I opted to change the return type to `List`, since the concept of an identity `Set` technically violates the `Set` contract.

We could alternatively use GDX's IdentityMap directly and just work with its keys since the field is private and we only interact with it minimally. Not sure if it would perform slightly better than the IdentityHashMap. But it's not likely to be frequently accessed in the game loop anyway, so maybe there's no point in adding that complexity.